backend/Controllers/SightingsController.cs

```diff
@@ -68,9 +70,72 @@ public ActionResult<List<ExtendedSightingResponse>> GetAllSightings()
                         Email = s.CreatedBy.Email,
                         Username = s.CreatedBy.Username
                     }
-            }).ToList();
+            });
+
+            //All 3 are selected
+            if (SearchTerm.LocationId != null && SpeciesSearchTerm.SpeciesId != null && UserSearchTerm.CreatedByUserId  != null) {
+             return results
+                    .Where(s => s.Location.Id == SearchTerm.LocationId)
+                    .Where(s => s.Species.Id == SpeciesSearchTerm.SpeciesId)
+                    .Where(s => s.User.Id == UserSearchTerm.CreatedByUserId)
+                    .ToList();
```

Comment:
Oh gosh there's a _lot_ of code here!

I would give something like this a go and see what happens. Linq might be smart enough to optimise this:
```suggestion
             return results
                    .Where(s => SearchTerm.LocationId == null || s.Location.Id == SearchTerm.LocationId)
                    .Where(s => SpeciesSearchTerm.SpeciesId == null || s.Species.Id == SpeciesSearchTerm.SpeciesId)
                    .Where(s => UserSearchTerm.CreatedByUserId == null || s.User.Id == UserSearchTerm.CreatedByUserId)
                    .ToList();
```